pkg/vault/test/foundry/fork/ERC4626WrapperBase.t.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { IERC4626 } from "@openzeppelin/contracts/interfaces/IERC4626.sol";
+
+abstract contract ERC4626WrapperBaseTest is Test {
+    // Variables to be defined by setUpForkTestVariables().
+    string internal network;
+    uint256 internal blockNumber;
+    IERC4626 internal wrapper;
+    address internal underlyingDonor;
+    uint256 internal amountToDonate;
+
+    IERC20 internal underlyingToken;
+    uint256 internal underlyingToWrappedFactor;
+
+    address internal user;
+    uint256 internal userInitialUnderlying;
+    uint256 internal userInitialShares;
+
+    uint256 internal constant MIN_DEPOSIT = 100;
+    // Tolerance of 1 wei difference between convert/preview and actual operation.
+    uint256 internal constant TOLERANCE = 1;
```

Comment:
Question: Why is the tolerance 1 wei here, but we use 2 wei in the mint path?

Reply:
Just because the mint path is in the vault, so I was afraid of being too restrictive, but 1 would work for all wrappers that we've tested so far.